### PR TITLE
HPCC-24991 Remove ldapAuthMethod option from ldap config

### PR DESCRIPTION
**Pull request by RussWhitehead**
Removed unused "kerberos" auth method from LDAP configuration

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@mayx Please review

**Comment by mayx** [comment]
@RussWhitehead is it possible to remove this option?

**Comment by mayx** [comment]
@RussWhitehead I meant can we remove the ldapAuthMethod config entry?

**Comment by RussWhitehead** [comment]
@mayx Removed ldapAuthMethod from ESP configuration. please rereview.

File: initfiles/componentfiles/configschema/xsd/dali.xsd [hunk 1]
```diff
@@ -95,15 +95,15 @@
                             </xs:restriction>
                         </xs:simpleType>
                     </xs:attribute>
-                    <xs:attribute name="authMethod" type="xs:string" hpcc:presetValue="kerberos"
+                    <xs:attribute name="authMethod" type="xs:string" hpcc:presetValue=""
```
Comment (mayx) [inline initfiles/componentfiles/configschema/xsd/dali.xsd:98]: Should the "authMethod" attribute be removed? Also LDAP/secmgrPlugin probably don't belong here...

File: initfiles/componentfiles/configxml/dali.xsd [hunk 2]
```diff
@@ -374,16 +374,16 @@
         </xs:restriction>
       </xs:simpleType>
     </xs:attribute>
-    <xs:attribute name="authMethod" use="optional" default="kerberos">
+    <xs:attribute name="authMethod" use="optional" default="">
```
Comment (mayx) [inline initfiles/componentfiles/configxml/dali.xsd:377]: Same as above, the authMethod attribute probably should be removed (please test dali deployment after the removal to make sure nothing breaks...)

**Review by mayx — COMMENTED** [review]
@RussWhitehead overall looks good, just a couple of comments related to dali.xsd.

**Comment by RussWhitehead** [comment]
@mayx The authMethod is required in Dali in order to specify to use LDAP or a pluggable security manager. I removed the "kerberos" option and tested in the original branch.

**Review by mayx — APPROVED** [review]
Looks good.

**Comment by RussWhitehead** [comment]
@richardkchapman Please merge

**Pull request merged**